### ISSUE TITLE
Add training visualization

### ISSUE THREAD

**Issue by RandomDefaultUser** [mala-project/mala#19]
In GitLab by @RandomDefaultUser on Dec 21, 2020, 08:44

Student project: Add training data visualization to the workflow. This includes (tasks not necesseray in optimal order): 

- Create and checkout your own branch (if your name is Firstname Lastname: YYMMDD_FL_AddTrainingVisualization)
- Identify and install the visualization framework of your choice (e.g. WandB, TensorBoard)
- Add installation of this framework to the installation guide
- Add a Visualizer class as visualizer.py in the network/ subdirectory
- Add a new Parameter subclass ParametersVisualization in parameter.py and make sure it gets called in the constructor of the Parameter class
    - Use this Parameter subclass to bundle all parameters you need during implementation
- Setup visualization in the Trainer class
    - either pass the visualization object in the constructor of Trainer or instantiate it in the constructor
    - visualize relevant data (training loss, validation loss, test loss)
    - add options for visualization to be passed by the ParametersVisualization class (such as "visualize_training_loss = False")
- Add an example_*.py file to showcase visualization and test it locally with downsized, preprocessed data
- Contact @fiedle09 to setup a training session on the hemera cluster and test it
- create a merge request, assign @fiedle09 - Done!
